Examples/Tests/Langmuir/analysis_langmuir_multi_2d.py

```diff
@@ -35,88 +42,116 @@
+t  = ds.current_time.to_value()
+data = ds.covering_grid(level=0, left_edge=ds.domain_left_edge, dims=ds.domain_dimensions)
+edge = np.array([np.array([(ds.domain_left_edge[1]).item(), (ds.domain_right_edge[1]).item(), \
+                           (ds.domain_left_edge[0]).item(), (ds.domain_right_edge[0]).item()])])
 
 # Check the validity of the fields
-error_rel = 0
+error_rel = 0.
 for field in ['Ex', 'Ez']:
     E_sim = data[field].to_ndarray()[:,:,0]
-    E_th = get_theoretical_field(field, t0)
-    max_error = abs(E_sim-E_th).max()/abs(E_th).max()
-    print('%s: Max error: %.2e' %(field,max_error))
-    error_rel = max( error_rel, max_error )
-
-# Plot the last field from the loop (Ez at iteration 40)
-plt.subplot2grid( (1,2), (0,0) )
-plt.imshow( E_sim )
-plt.colorbar()
-plt.title('Ez, last iteration\n(simulation)')
-plt.subplot2grid( (1,2), (0,1) )
-plt.imshow( E_th )
-plt.colorbar()
-plt.title('Ez, last iteration\n(theory)')
-plt.tight_layout()
-plt.savefig('langmuir_multi_2d_analysis.png')
-
-tolerance_rel = 0.05
-
-print("error_rel    : " + str(error_rel))
-print("tolerance_rel: " + str(tolerance_rel))
-
-assert( error_rel < tolerance_rel )
+    E_th = get_theoretical_field(field, t)
+    max_error = abs(E_sim - E_th).max() / abs(E_th).max()
+    print('{:s}: max error = {}'.format(field, max_error))
+    error_rel = max(error_rel, max_error)
+
+tolerance = 0.05
+print("error_rel = {}".format(error_rel))
+print("tolerance = {}".format(tolerance))
+assert(error_rel < tolerance)
+
+# Plot Ez
+E_sim = data['Ez'].to_ndarray()[:,:,0]
+
+fig, (ax1,ax2) = plt.subplots(1,2, dpi=100)
+# Absolute min and max field values
+vmin = min(E_sim.min(), E_th.min())
+vmax = max(E_sim.max(), E_th.max())
+# First plot
+cax1 = make_axes_locatable(ax1).append_axes('right', size='5%', pad='5%')
+im1 = ax1.imshow(E_sim, origin = 'lower', extent = edge[0], vmin = vmin, vmax = vmax)
+cb1 = fig.colorbar(im1, cax = cax1)
+ax1.set_xlabel(r'$z$')
+ax1.set_ylabel(r'$x$')
+ax1.xaxis.set_major_locator(plt.MaxNLocator(5))
+ax1.yaxis.set_major_locator(plt.MaxNLocator(5))
+ax1.set_title('Ez (sim)')
+cb1.ax.tick_params()
+cb1.ax.yaxis.get_offset_text().set()
+cb1.ax.yaxis.offsetText.set_ha('center')
+cb1.ax.yaxis.offsetText.set_va('bottom')
+cb1.formatter.set_powerlimits((0,0))
+cb1.update_ticks()
+# Second plot
+cax2 = make_axes_locatable(ax2).append_axes('right', size='5%', pad='5%')
+im2 = ax2.imshow(E_th, origin = 'lower', extent = edge[0], vmin = vmin, vmax = vmax)
+cb2 = fig.colorbar(im2, cax = cax2)
+ax2.set_xlabel(r'$z$')
+ax2.set_ylabel(r'$x$')
+ax2.xaxis.set_major_locator(plt.MaxNLocator(5))
+ax2.yaxis.set_major_locator(plt.MaxNLocator(5))
+ax2.set_title('Ez (theory)')
+cb2.ax.tick_params()
+cb2.ax.yaxis.get_offset_text().set()
+cb2.ax.yaxis.offsetText.set_ha('center')
+cb2.ax.yaxis.offsetText.set_va('bottom')
+cb2.formatter.set_powerlimits((0,0))
+cb2.update_ticks()
+# Save figure
+fig.tight_layout()
+fig.savefig('analysis_langmuir_multi_2d.png', dpi=200)
```

Comment:
This new code should produce better plots. Below is a comparison of the plots produced by the previous analysis script and the current one.

Before this PR:
<p align="center">
    <img src="https://user-images.githubusercontent.com/59625522/108256176-0c05f880-7112-11eb-8980-162ebfbeadf4.png" width="400">
</p>

After this PR:
<p align="center">
    <img src="https://user-images.githubusercontent.com/59625522/108256231-1e803200-7112-11eb-9ef8-2608aec169cd.png" width="400">
</p>